mir_eval/alignment.py

```diff
@@ -0,0 +1,300 @@
+    reference_timestamps : np.ndarray
+        reference timestamps, in seconds
+    estimated_timestamps : np.ndarray
+        estimated timestamps, in seconds
+    window : float
+        Window size, in seconds
+        (Default value = .1)
+    Returns
+    -------
+    pc : float
+        Percentage of correct timestamps
+    """
+    validate(reference_timestamps, estimated_timestamps)
+    deviations = np.abs(reference_timestamps - estimated_timestamps)
+    return np.mean(deviations <= window)
+
+
+def pcs(reference_timestamps, estimated_timestamps, duration: float):
+    """Constructs segments out of predicted and estimated timestamps separately
+    out of each given timestamp vector with entries (t1,t2, ... tN), yielding segments with
+    the following (start, end) boundaries: (0, t1), (t1, t2), ... (tN, duration).
+    The metric then calculates the percentage of overlap between correct segments compared to the
+    total duration.
+
+    This metric is based on the paper
+    "LyricSynchronizer: Automatic synchronization system between musical audio signals and lyrics"
+
+    Examples
+    --------
+    >>> reference_timestamps = mir_eval.io.load_events('reference.txt')
+    >>> estimated_timestamps = mir_eval.io.load_events('estimated.txt')
+    >>> duration = max(np.max(reference_timestamps), np.max(estimated_timestamps)) + 10
+    >>> pcs = mir_eval.align.pcs(reference_onsets, estimated_timestamps, duration)
+
+    Parameters
+    ----------
+    reference_timestamps : np.ndarray
+        reference timestamps, in seconds
+    estimated_timestamps : np.ndarray
+        estimated timestamps, in seconds
+    duration : float
+        Total duration of audio (seconds) - this is needed to compute the final segments properly!
+    Returns
+    -------
+    pcs : float
+        Percentage of time where ground truth and predicted segments overlap
+    """
+    validate(reference_timestamps, estimated_timestamps)
+    duration = float(duration)
+    assert np.max(reference_timestamps) <= duration, (
+        "Expected largest reference timestamp "
+        f"{np.max(reference_timestamps)} to not be "
+        f"larger than duration {duration}"
+    )
+    assert np.max(estimated_timestamps) <= duration, (
+        "Expected largest estimate timestamp "
+        f"{np.max(estimated_timestamps)} to not be "
+        f"larger than duration {duration}"
+    )
+
+    ref_starts = np.concatenate([[0], reference_timestamps])
+    ref_ends = np.concatenate([reference_timestamps, [duration]])
+    est_starts = np.concatenate([[0], estimated_timestamps])
+    est_ends = np.concatenate([estimated_timestamps, [duration]])
+    overlap_starts = np.maximum(ref_starts, est_starts)
+    overlap_ends = np.minimum(ref_ends, est_ends)
+    overlap_duration = np.sum(np.maximum(overlap_ends - overlap_starts, 0))
+    return overlap_duration / duration
+
+
+def perceptual_synchrony(offset):
+    """function perceptually weighting the deviations of onsets"""
```

Comment:
Please mention the ref for these values in this docstring too.

Reply:
Function doesn't exist anymore so that's resolved